lib/service_actor/conditionable.rb

```diff
@@ -13,26 +13,66 @@
 #             exist: -> provider { PROVIDERS.include?(provider) },
 #           }
 #   end
+#
+#   class Pay < Actor
+#     input :provider,
+#           must: {
+#             exist: {
+#               is: -> provider { PROVIDERS.include?(provider) },
+#               message: (lambda do |_input_key, _check_name, value|
+#                 "The specified provider \"#{value}\" was not found."
+#               end)
+#             }
+#           }
+#   end
 module ServiceActor::Conditionable
   def self.included(base)
     base.prepend(PrependedMethods)
   end
 
   module PrependedMethods
-    def _call
+    DEFAULT_MESSAGE = lambda do |input_key, check_name, value|
+      "Input #{input_key} must #{check_name} but was #{value.inspect}"
+    end
+
+    private_constant :DEFAULT_MESSAGE
+
+    def _call # rubocop:disable Metrics/MethodLength
       self.class.inputs.each do |key, options|
         next unless options[:must]
 
-        options[:must].each do |name, check|
+        options[:must].each do |check_name, check|
           value = result[key]
+
+          base_arguments = {
+            input_key: key,
+            check_name: check_name,
+            value: value
+          }
+
+          check, message = define_check_from(check)
+
           next if check.call(value)
 
-          raise ServiceActor::ArgumentError,
-                "Input #{key} must #{name} but was #{value.inspect}"
+          raise_error_with(message, **base_arguments)
         end
       end
 
       super
     end
+
+    private
+
+    def define_check_from(check)
+      message = DEFAULT_MESSAGE
+
+      # advanced mode
+      check, message = check.values_at(:is, :message) if check.is_a?(Hash)
+
+      [
+        check,
+        message
+      ]
```

Comment:
With an if/else we can slim this down too:

```suggestion
      if check.is_a?(Hash)
        check.values_at(:is, :message)
      else
        [check, DEFAULT_MESSAGE]
      end
```